code/modules/projectiles/guns/energy/special.dm

```diff
@@ -241,9 +249,16 @@
 			if(istype(WH))
 				WH.gun = WEAKREF(src)
 
-/obj/item/gun/energy/wormhole_projector/process_chamber()
-	..()
-	select_fire()
+/obj/item/gun/energy/wormhole_projector/afterattack(atom/target, mob/living/user, flag, params)
+	var/list/modifiers = params2list(params)
+	if(!LAZYACCESS(modifiers, RIGHT_CLICK) && select == AMMO_SELECT_ORANGE) //left click to fire blue wormholes.
+		select_fire()
+	return ..()
+
+/obj/item/gun/energy/wormhole_projector/afterattack_secondary(atom/target, mob/living/user, flag, params)
+	if(select == AMMO_SELECT_BLUE) //right click to fire blue wormholes.
+		select_fire()
+	return ..()
```

Comment:
This is wrong.

You can `..()` in the beginning, that's fine, but at the end you need to return:

```suggestion
	return SECONDARY_ATTACK_CONTINUE_CHAIN
```

The default is `SECONDARY_ATTACK_CALL_NORMAL`, which is only there for backwards compatibility reasons and is never what you want.